### PR TITLE
Added telemetry configuration in ALL

### PR DESCRIPTION
**Pull request by sudeeppr1998**
Added telemetry configuration in ALL

### CODE REVIEW

File: src/services/telementryService.js [hunk 1]
```diff
@@ -56,43 +56,54 @@ export const initialize = ({ context, config, metadata }) => {
...
-  CsTelemetryModule.instance.telemetryService.raiseStartTelemetry({
-    options: getEventOptions(),
-    edata: {
-      type: 'content',
-      mode: 'play',
-      stageid: url,
-      duration: Number((duration / 1e3).toFixed(2)),
-    },
-  });
+  if (process.env.REACT_APP_TELEMETRY_MODE === 'ET' || process.env.REACT_APP_TELEMETRY_MODE === 'NT' || process.env.REACT_APP_TELEMETRY_MODE === 'DT') {
```
Comment (gouravmore) [inline src/services/telementryService.js:59]: START and END event are always required so you can remove the check from here

File: src/services/telementryService.js [hunk 2]
```diff
@@ -56,43 +56,54 @@ export const initialize = ({ context, config, metadata }) => {
...
-  CsTelemetryModule.instance.telemetryService.raiseEndTelemetry({
-    edata: {
-      type: 'content',
-      mode: 'play',
-      pageid: url,
-      summary: [],
-      duration: '000',
-    },
-  });
+  if (process.env.REACT_APP_TELEMETRY_MODE === 'ET' || process.env.REACT_APP_TELEMETRY_MODE === 'NT' || process.env.REACT_APP_TELEMETRY_MODE === 'DT') {
```
Comment (gouravmore) [inline src/services/telementryService.js:86]: START and END event are always required so you can remove the check from here

File: src/services/callTelemetryIntract.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import { interact,start } from '../services/telementryService';
 const duration = new Date().getTime();
-export const interactCall = () => {
-  interact()
+export const interactCall = (telemetryMode,id) => {
```
Comment (gouravmore) [inline src/services/callTelemetryIntract.js:4]: are we using this id param anywhere?

File: src/services/telementryService.js [hunk 4]
```diff
@@ -56,43 +56,54 @@ export const initialize = ({ context, config, metadata }) => {
...
 };
 
 export const response = (context, options) => {
-  CsTelemetryModule.instance.telemetryService.raiseResponseTelemetry(
-    {
-      ...context,
-    },
-    getEventOptions()
-  );
+  if (process.env.REACT_APP_TELEMETRY_MODE === 'ET' || process.env.REACT_APP_TELEMETRY_MODE === 'NT' || process.env.REACT_APP_TELEMETRY_MODE === 'DT') {
```
Comment (gouravmore) [inline src/services/telementryService.js:74]: need to update the login to => 
 (this.TELEMETRY_MODE === 'ET' && currentMode === 'ET') ||
      (this.TELEMETRY_MODE === 'NT' &&
        (currentMode === 'ET' || currentMode === 'NT')) ||
      (this.TELEMETRY_MODE === 'DT' &&
        (currentMode === 'ET' || currentMode === 'NT' || currentMode === 'DT'))
    )

File: src/services/telementryService.js [hunk 5]
```diff
@@ -56,43 +56,54 @@ export const initialize = ({ context, config, metadata }) => {
...
+  }
+
 };
 
-export const interact = id => {
-  CsTelemetryModule.instance.telemetryService.raiseInteractTelemetry({
-    options: getEventOptions(),
-    edata: { type: 'TOUCH', subtype: '', id, pageid: url },
-  });
+export const interact = (telemetryMode,id) => {
```
Comment (gouravmore) [inline src/services/telementryService.js:100]: The 'id' should be the first parameter that is required, while optional parameters should be placed at the end in a functional declaration.

File: src/services/telementryService.js [hunk 6]
```diff
@@ -112,33 +123,39 @@ export const search = id => {
...
 export const feedback = (data, contentId) => {
-  CsTelemetryModule.instance.telemetryService.raiseFeedBackTelemetry({
-    options: getEventOptions(),
-    edata: {
-      contentId: contentId,
-      rating: data,
-      comments: '',
-    },
-  });
+  if (process.env.REACT_APP_TELEMETRY_MODE === 'ET' || process.env.REACT_APP_TELEMETRY_MODE === 'NT' || process.env.REACT_APP_TELEMETRY_MODE === 'DT') {
```
Comment (gouravmore) [inline src/services/telementryService.js:149]: can we create one common function which will return true/false

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/services/callTelemetryIntract.js (hunk 3 above)
Comment (sudeeppr1998) [reply src/services/callTelemetryIntract.js:4]: No we are not storing it anywhere.

**Pull request merged**